cmd/nerdctl/ps.go

```diff
@@ -62,11 +65,46 @@ func psAction(cmd *cobra.Command, args []string) error {
 		return err
 	}
 	defer cancel()
-	containers, err := client.Containers(ctx)
+	all, err := cmd.Flags().GetBool("all")
+	if err != nil {
+		return err
+	}
+	latest, err := cmd.Flags().GetBool("latest")
 	if err != nil {
 		return err
 	}
-	return printContainers(ctx, cmd, containers)
+	lastN, err := cmd.Flags().GetInt("last")
+	if err != nil {
+		return err
+	}
+	if lastN == -1 && latest {
+		lastN = 1
+	}
+	containers := []containerd.Container{}
+	if lastN > 0 || lastN == -1 {
+		containers, err = client.Containers(ctx)
+		if err != nil {
+			return err
+		}
+	}
+	if lastN > 0 {
+		all = true
+		sort.Slice(containers, func(i, j int) bool {
+			infoI, err := containers[i].Info(ctx)
```

Comment:
Better to cache `Info()` result in another slice before calling `sort.Slice`

Reply:
Can just pass `WithoutRefreshedMetadata` option to use cache?
like https://github.com/containerd/nerdctl/pull/598/files#diff-2ffc2cdeb00bd7b916b01ebdf34c9083512cf0057fdd7d886b794e4d5e510eccR165

Reply:
SGTM

Reply:
I'll deal with it this way first, and optimize if necessary.